components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -627,6 +649,7 @@ private Application createOAuthApplication(ApplicationRegistrationRequest regist
         }
         Application application = buildResponse(createdApp, tenantDomain);
         application.setSoftwareStatement(registrationRequest.getSoftwareStatement());
+        application.setExtAllowedAudience(serviceProvider.getAssociatedRolesConfig().getAllowedAudience());
```

Comment:
can't we move this to 617 into the if clause?

Reply:
Same case as mentioned above.